File: src/backend/go/internal/dto/file.go
package dto

import "time"

// FileResponse mirrors a FileDto / view model returned from controllers.
// The internal storage `location` is intentionally not exposed to clients;
// file content is retrieved via the dedicated download endpoint instead.
type FileResponse struct {
	ID          uint       `json:"id"`
	Name        string     `json:"name"`
	Extension   string     `json:"extension"`
	Size        int64      `json:"size"`
	ContentType *string    `json:"contentType"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   *time.Time `json:"updatedAt"`
}

File: src/backend/python/api/schemas/file.py
from datetime import datetime

from pydantic import BaseModel


# ── Response DTO (analogous to a FileDto / view model) ─────────────────────────
# Note: the on-disk `location` is intentionally not exposed to clients; content is
# retrieved via the dedicated download endpoint instead.

class FileResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    name: str
    extension: str
    size: int
    content_type: str | None
    created_at: datetime
    created_by_user_id: int | None = None
    updated_at: datetime | None
    updated_by_user_id: int | None = None
